lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1443,12 +1467,19 @@ def initialize_environment(self, environment: Environment, log: Logger) -> None:
             public_address, private_address = get_primary_ip_addresses(
                 self, resource_group_name, vms_map[vm_name]
             )
+
+            connection_public_ip = node_context.public_ip_address
+            if self._azure_runbook.use_private_address_for_vm_connection:
+                # setting public_address to empty causes set_connection_info
+                # to use address (the private ip)
+                connection_public_ip = ''
```

Comment:
might make more sense to change this to use `node_context.private_ip_address` rather than an empty string.